packages/frontend-test-utils/src/app/createExtensionTester.test.tsx

```diff
@@ -14,55 +14,84 @@
-  it('should render a simple extension', async () => {
-    createExtensionTester(
-      createExtension({
-        namespace: 'test',
-        attachTo: { id: 'ignored', input: 'ignored' },
-        output: { element: coreExtensionData.reactElement },
-        factory: () => ({ element: <div>test</div> }),
-      }),
-    ).render();
+  const defaultDefinition = {
+    namespace: 'test',
+    attachTo: { id: 'ignored', input: 'ignored' },
+    output: { element: coreExtensionData.reactElement },
+    factory: () => ({ element: <div>test</div> }),
+  };
 
+  it('should render a simple extension', async () => {
+    const extension = createExtension(defaultDefinition);
+    const tester = createExtensionTester(extension);
+    tester.render();
     await expect(screen.findByText('test')).resolves.toBeInTheDocument();
   });
 
   it('should render an extension even if disabled by default', async () => {
-    createExtensionTester(
-      createExtension({
-        namespace: 'test',
-        attachTo: { id: 'ignored', input: 'ignored' },
-        disabled: true,
-        output: { element: coreExtensionData.reactElement },
-        factory: () => ({ element: <div>test</div> }),
-      }),
-    ).render();
-
+    const extension = createExtension({
+      ...defaultDefinition,
+      disabled: true,
+    });
+    const tester = createExtensionTester(extension);
+    tester.render();
     await expect(screen.findByText('test')).resolves.toBeInTheDocument();
   });
 
   it("should fail to render an extension that doesn't output a react element", async () => {
-    expect(() =>
-      createExtensionTester(
-        createExtension({
-          namespace: 'test',
-          attachTo: { id: 'ignored', input: 'ignored' },
-          disabled: true,
-          output: { path: coreExtensionData.routePath },
-          factory: () => ({ path: '/foo' }),
-        }),
-      ).render(),
-    ).toThrow(
-      "Failed to instantiate extension 'core/router', input 'children' did not receive required extension data 'core.reactElement' from extension 'test'",
+    const extension = createExtension({
+      ...defaultDefinition,
+      output: { path: coreExtensionData.routePath },
+      factory: () => ({ path: '/foo' }),
+    });
+    const tester = createExtensionTester(extension);
+    expect(() => tester.render()).toThrow(
+      "Failed to instantiate extension 'core/routes', input 'routes' did not receive required extension data 'core.reactElement' from extension 'test'",
     );
   });
+
+  it('should render multiple extensions', async () => {
+    const indexPageExtension = createExtension({
+      ...defaultDefinition,
+      factory: () => ({
+        element: (
+          <div>
+            Index page <Link to="/details">See details</Link>
+          </div>
+        ),
+      }),
+    });
+    const detailsPageExtension = createExtension({
+      ...defaultDefinition,
+      name: 'details',
+      attachTo: { id: 'core/routes', input: 'routes' },
```

Comment:
I have a feeling this test is fragile to break because we are passing the id string manually here. My impression is that we could have a variable to reference the id or use the page extension factory instead (since the test is asynchronous anyway).

Reply:
this is gonna be app/routes after #21902